padl/transforms.py

```diff
@@ -723,20 +766,29 @@ def _pd_itercall(self, args, mode: Mode, loader_kwargs: Optional[dict] = None,
         if use_preprocess:
             loader = self.pd_get_loader(args, preprocess, mode, **loader_kwargs)
         else:
-            loader = args
+            loader = list(zip(args, range(len(args))))
 
         def _gen():
-            for batch in loader:
+            for batch, ix in loader:
```

Comment:
Could we use `enumerate` here instead of using `list` and `zip` above?

Reply:
yes, that's better

Reply:
well, the problem with this is i need each element with the following order: (args, item) and enumerate gives it to me as (item, args)

Reply:
Why do you need it in the order (args, item)?

Reply:
A potential issue I see with using `list` is that if someone passed in `args` as an `iterator`, similar to our `_ItemGetter` is that it would force the `iterator` to be completely moved into memory. This could then cause problems if the `iterator` is too big to fit into memory.

Reply:
> Why do you need it in the order (args, item)?

yes, I did not want to change so much the code but it is a better call, I changed the order to `for ix, batch` in loader (and also the corresponding line in `_ItemGetter`)

Reply:
> A potential issue I see with using `list` is that if someone passed in `args` as an `iterator`, similar to our `_ItemGetter` is that it would force the `iterator` to be completely moved into memory. This could then cause problems if the `iterator` is too big to fit into memory.

I see, then it would be nice to use an iterator, but the problem with `enumerate` is that it does not have a `__len__` method, but we could use a customized one with that method, what do you think?

Reply:
Yeah I think it would be better to make a custom iterator to wrap around `args` to get it to do what you want.

Reply:
I've made a new iterator similar to `_ItemGetter`. Actually, it would also be an option to introduce an argument in `_ItemGetter`to act in two different ways.